deploy/zyxel_gs1900.sh

```diff
@@ -0,0 +1,423 @@
+  minimum_minor_version=$(_zyxel_gs1900_parse_minor_version "$_zyxel_gs1900_minimum_firmware_version")
+  _debug2 "minimum_minor_version" "$minimum_minor_version"
+
+  _debug2 "firmware_version" "$firmware_version"
+  firmware_major_version=$(_zyxel_gs1900_parse_major_version "$firmware_version")
+  _debug2 "firmware_major_version" "$firmware_major_version"
+  firmware_minor_version=$(_zyxel_gs1900_parse_minor_version "$firmware_version")
+  _debug2 "firmware_minor_version" "$firmware_minor_version"
+  
+  _ret=0
+  if [ "$firmware_major_version" -lt "$minimum_major_version" ]; then
+    _ret=1
+  elif [ "$firmware_major_version" -eq "$minimum_major_version" ] && [ "$firmware_minor_version" -lt "$minimum_minor_version" ]; then
+    _ret=1
+  fi
+
+  if [ "$_ret" != "0" ]; then
+    _err "Unsupported firmware version $firmware_version. Please upgrade to at least version $_zyxel_gs1900_minimum_firmware_version."
+  fi
+  
+  return $?
+}
+
+_zyxel_gs1900_upload_certificate() {
+  # Generate a PKCS12 certificate with a temporary password since the web interface
+  # requires a password be present. Then upload that certificate.
+  temp_cert_password=$(head /dev/urandom | tr -dc 'A-Za-z0-9' | head -c 64)
+  _secure_debug2 "tmp_cert_password" "$temp_cert_password"
+
+  temp_pkcs12="$(_mktemp)"
+  _debug2 "temp_pkcs12" "$temp_pkcs12"
+  _toPkcs "$temp_pkcs12" "$_ckey" "$_ccert" "$_cca" "$temp_cert_password"
+  if [ "$?" != "0" ]; then
+    _err "Failed to generate a pkcs12 certificate."
+    _err "Please re-run with --debug and report a bug."
+
+    # ensure the temporary certificate file is cleaned up
+    [ -f "${temp_pkcs12}" ] && rm -f "${temp_pkcs12}"
+
+    return $?
+  fi
+
+  upload_page_html=$(_get "${_zyxel_switch_base_uri}/cgi-bin/dispatcher.cgi?cmd=5914" | tr -d '\n')
+
+  # Get the two validity dates by looking for their date format in the page (i.e. Mar 5 05:48:48 2024 GMT)
+  existing_validity=$(_zyxel_html_extract_dates "$upload_page_html")
+  _debug2 "existing_validity" "$existing_validity"
+
+  form_xss_value=$(printf "%s" "$upload_page_html" | _egrep_o 'name="XSSID"\s*value="[^"]+"' | sed 's/^.*="\([^"]\{1,\}\)"$/\1/g')
+  _secure_debug2 "form_xss_value" "$form_xss_value"
+  
+  # If a certificate exists on the switch already there will be two XSS keys - we want the first one
+  form_xss_value=$(printf "%s" "$form_xss_value" | head -n 1)
+  _secure_debug2 "form_xss_value" "$form_xss_value"
+
+  _debug2 "Send upload request to the server"
+  _inithttp
+  _CURL="$_ACME_CURL"
+  if [ "$HTTPS_INSECURE" ]; then
+    _CURL="$_CURL --insecure  "
+  fi
+
+  # Unfortunately we cannot rely upon the switch response across switch models
+  # to return a consistent body return - so we cannot inspect the result of this
+  # upload to determine success. We will need to re-query the certificates page
+  # and compare the validity dates to try and identify if they have changed.
+  upload_response_html=$($_CURL \
```

Comment:
only for `curl` here?  how about `wget` ?
I would suggest you not use `curl` here,  please use `_post()` method instead.

Reply:
I originally followed the example in `synology_dsm.sh` and constructed the multpart form request by hand. This did not work in either `_post()` or `wget`. Both of those mangled the binary certificate data.

Specifically some of the shell-unsafe bytes and null bytes seemed to be getting mangled.

I am willing to try again, but I did not have success originally.

Reply:
I will try, but I'm pretty sure the server is expecting unencoded binary data for the certificate file in the multi part form upload.

Reply:
Thank you for your suggestion. My computer has broken and I'm now waiting on replacement parts.

Once I'm back up and running I will be trying what you suggest.

Reply:
I have tried your suggestion but I cannot make it work.
I will am updating the PR to draft, and updating the branch to see if you have any other ideas to avoid the `curl` requirement.